### DISCUSSION TITLE
Metal Support

### DISCUSSION THREAD

**Discussion by mousebird** [maplibre/maplibre-native#202] [Q&A]
I see you've made use of ANGLE to get Metal compatibility.  Have you considered a full Metal port?
That would consist of native Metal shaders and, presumably, an indirect mode implementation.

I'm asking for a client that's considering their options.

For the record we (mousebird consulting) do make another toolkit, but it's not on the table.  The client is really, really invested in the MB/GL stack and we've been pulled in to advise and consult.

If you're not clear on why you'd want a direct Metal port, I can pull together some traces to show you.

**Comment by wipfli** [comment]
Hi @mousebird I think there is considerable interest in getting Metal support in MapLibre GL JS. If people were to starting discussing and implementing Metal support, it would be hugely beneficial to get expert feedback and inputs. Would you or your customer still be interested in joining such an effort?

**Reply by sjg-wdw** [reply]
They're paying attention, but I'm not sure where their head is at currently.

My main comment about Metal support is that it's not a volunteer kind of thing.  It needs sustained effort in the six figure range for a decent amount of time.  At the same time I'd also recommend a lot of work to bring MapLibre up to date with modern practices unrelated to rendering.

Getting that money is the hardest part and herding those cats is going to be almost impossible.  I wish you the best of luck!

If you've got some specific effort going on, I'll be happy to rope them in if they're not already aware of it.  Or if you have some big company who needs the situation explained, I'm happy to do that for free.  On the technical side, I'm not really in this business myself anymore.

**Reply by wipfli** [reply]
Thanks for the feedback @sjg-wdw

**Reply by sjg-wdw** [reply]
Sorry, that may have been confusing.  @mousebird and @sjs-wdw are both me.  New company.

**Reply by wipfli** [reply]
Yeah I saw that you do weather apps now. Sounds like a really cool business! Good luck with the meteo stuff :)
